src/wagtail_factories/factories.py

```diff
@@ -146,3 +147,31 @@ class Meta:
 
     title = "A document"
     file = factory.django.FileField()
+
+
+class RedirectFactory(DjangoModelFactory):
+    old_path = factory.Faker("slug")
+    site = factory.SubFactory(SiteFactory)
+    redirect_page = factory.SubFactory(PageFactory)
+    redirect_link = factory.Faker("url")
+    is_permanent = factory.Faker("boolean")
+
+    class Meta:
+        model = Redirect
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        """
+        Override _create() to ensure that Redirect.clean() is run
+        in order to normalise `old_path`.
+        """
+        obj = model_class(*args, **kwargs)
```

Comment:
Question: can we (or should we) defer creation of the instance to the superclass implementation?

Reply:
According to the [factoryboy documentation](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory._create) this is the recommended way of overriding `_create()`